### ISSUE TITLE
[a11y]: ai-chat may add aria relationship from links/references to container message

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#910]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

Imagine a response message that contains a text plus a number of references/links, i.e. a linkList element.
Imagine further, that one of the references/links has a very short title, like "Overview". This may occur because the titles may be taken from online documentation for example.

Such existing Label "Overview" would be too vague or insufficiently descriptive, to explain the purpose of the control .

When the references/links use aria-labelled-by or aria-owns to refer to the overall/containing message, the context would provided enough context. 

User impact: When controls have unclear or non-descriptive labels, users—particularly those using assistive technologies—may misunderstand the purpose of the control, leading to errors, hesitation, or inability to complete tasks.

### WCAG 2.1 Violation

WCAG Mapping: 2.4.6 Headings and Labels

### Reproduction/example

n/a

### Steps to reproduce

1. open ai-chat
2. ask question that causes an answer to contain references with insufficiently descriptive titles.
3. Observe with JAWS, that the context is insufficient

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
